certora/specs/Timelock.spec

```diff
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+import "LastUpdated.spec";
+
+methods {
+    function _.supplyShares(MetaMorphoHarness.Id id, address user) external => summarySupplyshares(id, user) expect uint256;
+}
+
+ghost lastSupplyShares(MetaMorphoHarness.Id, address) returns uint256;
+
+function summarySupplyshares(MetaMorphoHarness.Id id, address user) returns uint256 {
+    uint256 res;
+    require lastSupplyShares(id, user) == res;
+    return res;
+}
+
+// Show that nextGuardianUpdateTime does not revert.
+rule nextGuardianUpdateTimeDoesNotRevert() {
```

Comment:
> why do we want to prove this? Does it prove in some way that there exist a time in the future at which it can change?

No, it's just to check that we did not constrain the verification at all by calling this function. If it had a `require(false)` inside for example, we would be able to prove anything we want. Here it's simply a getter that never reverts, so it does not make more assumptions by calling it

Reply:
ah indeed! that's quite tricky

why btw having written these functions in solidity then?

Reply:
I thought it would be nice to have those functions in solidity as it's easier to show them. Like you could show them to users, and say: "run this function and you will know how long you are guaranteed that this value does not change". If this is not interesting I can write them in CVL (but it's a bit uglier, because we don't have `Math` and we can't change values in place)

Reply:
I see, let's keep like that